transformer_engine/pytorch/csrc/userbuffers/userbuffers-host.cpp

```diff
@@ -279,7 +282,14 @@ int create_communicator_grouped2(communicator **comm, int pipegpus, int pipenode
                            (NVTE_MAX_SMS + 100) * sizeof(int)));
   for (int i = 0; i < 100 + NVTE_MAX_SMS; i++)
     (*comm)->hostflags[i] = 0;
+#if (defined(__arm__) && __ARM_ARCH >= 7) || defined(__aarch64__)
```

Comment:
Few comments here:
* If you are writing to external device memory (potentially) this barrier is insufficient.
* SY barrier is very expensive and it is not clear if SY is required
* This should be a standalone macro

Reply:
sleep(1) is a good sign that it is not working :-) I don't know why it was added but this is no good. I believe `asm volatile("dsb st" ::: "memory");` is the right barrier. Can you try it ?

Reply:
@aurianer  After further review of the code we came to conclusion that the code is not used at all and should be dropped all together. I will add cleanup commit to #757